ChildProject/pipelines/metrics.py

```diff
@@ -80,46 +120,266 @@ def __init__(
+        
+        :return: DataFrame of computed metrics
+        :rtype: pandas.DataFrame
+        """
+        if self.threads == 1:
+            self.metrics = pd.DataFrame(
+                [self._process_unit(row) for row in self.metrics.iterrows()]
+            )
+        else:
+            with mp.Pool(
+                processes=self.threads if self.threads >= 1 else mp.cpu_count()
+            ) as pool:
+                self.metrics = pd.DataFrame(
+                    pool.map(self._process_unit, self.metrics.iterrows())
+                )
+        if self.period:
+            self.metrics['period_start'] = self.metrics['period_start'].dt.strftime('%H:%M:%S')
+            self.metrics['period_end'] = self.metrics['period_end'].dt.strftime('%H:%M:%S')
+        return self.metrics
 
-    def retrieve_segments(self, sets: List[str], unit: str):
-        annotations = self.am.annotations[self.am.annotations[self.by] == unit]
+    def retrieve_segments(self, sets: List[str], row: str):
+        """from a list of sets and and a row identifying the unit computed, return the relevant annotation segments
+        
+        :param sets: List of annotation sets to keep
+        :type sets: List[str]
+        :param row: Series storing the unit to compute information
+        :type row: pandas.Series
+        :return: relevant annotation DataFrame and index DataFrame
+        :rtype: (pandas.DataFrame , pandas.DataFrame)
+        """
+        annotations = self.am.annotations[self.am.annotations[self.by] == row[self.by]]
         annotations = annotations[annotations["set"].isin(sets)]
-
+        
         if self.from_time and self.to_time:
-            annotations = self.am.get_within_time_range(
-                annotations, self.from_time, self.to_time, errors="coerce"
-            )
+            if self.period:
+                st_hour = row["period_start"]
+                end_hour = row["period_end"]
+                intervals = time_intervals_intersect(TimeInterval(self.from_time,self.to_time),TimeInterval(st_hour,end_hour))
+                matches = pd.concat([self.am.get_within_time_range(annotations,i) for i in intervals],ignore_index =True) if intervals else pd.DataFrame()
+            else:
+                matches = self.am.get_within_time_range(
+                annotations, TimeInterval(self.from_time,self.to_time))
+        elif self.period:
+            st_hour = row["period_start"]
+            end_hour = row["period_end"]
+            matches = self.am.get_within_time_range(
+                    annotations, TimeInterval(st_hour,end_hour))
+        else:
+            matches = annotations
 
         try:
-            segments = self.am.get_segments(annotations)
+            segments = self.am.get_segments(matches)
         except Exception as e:
-            print(str(e))
+            #print(str(e))
+            #no annotations for that unit
             return pd.DataFrame(), pd.DataFrame()
 
         # prevent overflows
```

Comment:
(Obviously, has nothing with the new pipeline, so feel free to ignore it completely) Why is there a risk of overflow? Is that because the total duration might be too big an int? Wouldn't it be less than the maximum offset though?

Reply:
To be honest, I don't know any better. I checked the commits and that comment was already present in the first version of the file, with almost the same code. As for reaching a too high int value, I can't see that happening as I think that corresponds to weeks of continuous recordings...